service/src/main/java/io/techery/janet/http/model/MultipartRequestBody.java

```diff
@@ -1,49 +1,139 @@
 package io.techery.janet.http.model;
 
+import java.io.ByteArrayInputStream;
 import java.io.ByteArrayOutputStream;
 import java.io.IOException;
+import java.io.InputStream;
 import java.io.OutputStream;
+import java.io.SequenceInputStream;
 import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Collections;
 import java.util.LinkedList;
 import java.util.List;
 import java.util.UUID;
 
 import io.techery.janet.body.ActionBody;
 
 public final class MultipartRequestBody extends ActionBody {
+
+    public static final String MIMETYPE_FORM_DATA = "multipart/form-data";
     public static final String DEFAULT_TRANSFER_ENCODING = "binary";
 
+    private static final String DASH_DASH = "--";
+    private static final String CRLF = "\r\n";
+
+    private final List<MimePart> mimeParts = new LinkedList<MimePart>();
+
+    private final String boundary;
+    private final byte[] footer;
+    private long length;
+
+    public MultipartRequestBody() {
+        this(UUID.randomUUID().toString());
+    }
+
+    MultipartRequestBody(String boundary) {
+        super(MIMETYPE_FORM_DATA + "; boundary=" + boundary); //TODO add support for other mime types
+        this.boundary = boundary;
+        footer = buildBoundary(boundary, false, true);
+        length = footer.length;
+    }
+
+    @Override
+    public long length() {
+        return length;
+    }
+
+    @Override
+    public InputStream getContent() throws IOException {
+        List<InputStream> streams = new ArrayList<InputStream>(mimeParts.size());
+        for (MimePart mimePart : mimeParts) streams.add(mimePart.getContent());
+        SequenceInputStream streamsSequence = new SequenceInputStream(Collections.enumeration(streams));
+        return new SequenceInputStream(streamsSequence, new ByteArrayInputStream(footer));
+    }
+
+    @Override public void writeContentTo(OutputStream os) throws IOException {
```

Comment:
What do u think if this method will use the method `getContent()` as source to write data to output stream. It would provide single responsible method to safe data structure.